mm2src/coins/z_coin/z_params/indexeddb.rs

```diff
@@ -0,0 +1,172 @@
+        self.0
+            .get_or_initialize()
+            .await
+            .mm_err(|err| ZcoinStorageError::DbError(err.to_string()))
+    }
+
+    pub async fn save_params(
+        &self,
+        sapling_spend_id: u8,
+        sapling_spend: &[u8],
+        sapling_output: &[u8],
+    ) -> MmResult<(), ZcoinStorageError> {
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let params_db = db_transaction.table::<ZcashParamsWasmTable>().await?;
+        let params = ZcashParamsWasmTable {
+            sapling_spend_id,
+            sapling_spend: sapling_spend.to_vec(),
+            sapling_output: sapling_output.to_vec(),
+            ticker: "z_params".to_string(),
+        };
+
+        params_db.add_item(&params).await?;
+
+        Ok(())
+    }
+
+    pub async fn check_params(&self) -> MmResult<bool, ZcoinStorageError> {
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let params_db = db_transaction.table::<ZcashParamsWasmTable>().await?;
+
+        let maybe_param = params_db
+            .cursor_builder()
+            .only("ticker", "z_params")?
+            .open_cursor("ticker")
+            .await?
+            .next()
+            .await?;
+
+        Ok(maybe_param.is_some())
+    }
+
+    pub async fn get_params(&self) -> MmResult<(Vec<u8>, Vec<u8>), ZcoinStorageError> {
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let params_db = db_transaction.table::<ZcashParamsWasmTable>().await?;
+        let mut maybe_params = params_db
+            .cursor_builder()
+            .only("ticker", "z_params")?
+            .open_cursor("ticker")
+            .await?;
+
+        let mut sapling_spend = vec![];
+        let mut sapling_output = vec![];
+
+        while let Some((_, params)) = maybe_params.next().await? {
+            sapling_spend.extend_from_slice(&params.sapling_spend);
+            if params.sapling_spend_id == 0 {
+                sapling_output = params.sapling_output
+            }
+        }
+
+        Ok((sapling_spend, sapling_output.clone()))
+    }
+
+    pub async fn download_and_save_params(&self) -> MmResult<(Vec<u8>, Vec<u8>), ZcoinStorageError> {
+        let (sapling_spend, sapling_output) = super::download_parameters().await.unwrap();
+        let spends = sapling_spend_to_chunks(&sapling_spend);
+        for (i, spend) in spends.into_iter().enumerate() {
+            if i == 0 {
+                self.save_params(i as u8, &spend, &sapling_output).await?
+            } else {
+                self.save_params(i as u8, &spend, &[]).await?
+            }
+        }
+
+        Ok((sapling_spend, sapling_output))
+    }
+}
+
+pub fn sapling_spend_to_chunks(sapling_spend: &[u8]) -> Vec<Vec<u8>> {
```

Comment:
please add doc comment
UPD: also recheck please, may be you can leave it as pub(crate)

Reply:
done